### ISSUE TITLE
_value is weird as the name of a public data member of success

### ISSUE THREAD

**Issue by viboes** [ned14/outcome#93]
See https://ned14.github.io/outcome/reference/success_failure/#struct-outcome-v2-xxx-success-type

If the data member _value is public, why to have the value() accessor?

What about 
```
template <class T>
struct success_type
{
    using value_type = T;
    value_type value;
};
```

I know that std::unexpected do something like that, but the data member is not public.
I'll propose this simpler definition in the next paper.

**Comment by ned14** [comment]
You're right, originally the success and failure types were aggregate initialised types with C++ 17 template deduction guides and a C++ 14 emulation. But I could not get `success()` to work right due to compiler limitations, so I gave up and reverted to a pure C++ 14 solution.

Which I obviously did not completely finish off, so marking this as bug so I go finish the job properly. Thanks Vicente.

**Comment by ned14** [comment]
Fixed. Thanks for the bug report.

**Comment by viboes** [comment]
Could you summarize what was the fix here?

**Comment by ned14** [comment]
Access to struct members was removed, and member functions now provide observation. Constructors were added to allow aggregrate initialisation to continue to work.
